### PR TITLE
Add utilities

### PR DESCRIPTION
**Pull request by McCallisterRomer**
Adding:
- ScopeGuard
- Helper algorithm for std::transform
- Enumerate algorithm

### CODE REVIEW

File: include/ncutility/Algorithm.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+#include <algorithm>
+#include <type_traits>
+#include <vector>
+
+namespace nc::algo
+{
+/** @brief Wrapper around std::transform that returns transformed data in a new vector. */
+template<class T, std::invocable<const T&> UnaryOperation>
+auto Transform(const std::vector<T>& container, UnaryOperation op)
```
Comment (jaremieromer) [inline include/ncutility/Algorithm.h:13]: HOORAY!

File: include/ncutility/detail/EnumerateDetail.h [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+template<class R>
+concept simple_view =
+    std::ranges::view<R> &&
+    std::ranges::range<const R> &&
+    std::same_as<std::ranges::iterator_t<R>, std::ranges::iterator_t<const R>> &&
+    std::same_as<std::ranges::sentinel_t<R>, std::ranges::sentinel_t<const R>>;
+
+template<std::ranges::input_range V>
+    requires std::ranges::view<V>
+class enumerate_view : public std::ranges::view_interface<enumerate_view<V>> {
```
Comment (jaremieromer) [inline include/ncutility/detail/EnumerateDetail.h]: How are you determining curly spacing re: newlines? I think it should be on a new line for class declarations

**Review by jaremieromer — APPROVED** [review]
These are great!
In general, I'm curious about what scheme you're following for curlies re: newlines

File: include/ncutility/detail/EnumerateDetail.h (hunk 2 above)
Comment (McCallisterRomer) [reply include/ncutility/detail/EnumerateDetail.h]: Moved. This guy is from rangesnext. Missed that one.

**Pull request merged**